### DISCUSSION TITLE
Plan for v2.0 and new PRs?

### DISCUSSION THREAD

**Discussion by ccaprani** [robbievanleeuwen/section-properties#75] [General]
Hi @robbievanleeuwen @connorferster @Agent6-6-6 @Czarified @Spectre5 

I was planning to merge a set of standard bridge beams as part of the Common Structural Sections library. I've just caught up with the development conversations and am wondering is it better to wait until v2.0 at this stage. I already see there are a number of conflicts between the big PRs in the works and don't want to contribute confusion. 

Is there a release roadmap?

A secondary consideration is that the Common Structural Sections library is already pretty big, and could get a lot bigger. Is there a better design solution? For example, the library could be its own sub-package so that similar sections are collected in their own modules, e.g. `sectionproperties.pre.sections.library.bridgebeams`, `sectionproperties.pre.sections.library.steelsections`, and so on?

Cheers

**Comment by connorferster** [comment]
Hi @ccaprani

In regards to a release roadmap, no, nothing formal was ever discussed. I was intending for this project to be my "Christmas break project" of 2020 and has turned into something much more. The work is wrapping up and I am spending time revising the documentation and fine tuning some of the plotting and error catching.

There is currently a conflict with another PR but I just had an idea today for how to resolve some of it. 

Your bridge sections would be a welcome addition. I agree, I think that we should think about moving the section generating functions to a "library" module or something. 

For compatibility with v2.0, if you wrote a function to generate the geometry as a shapely polygon, it will be easy to plug in. Generating shapely polygons only requires ordered lists of coordinates, one for the exterior and one each of any interior holes (as opposed to nodes, facets, control points, and holes). We would appreciate any contribution(s) you are willing to write.

I am spending time on sectionproperties this weekend and hope to have more material for review shortly.

**Reply by ccaprani** [reply]
@connorferster sounds good, and glad to see thing moving along. You can see what I've got [here](https://github.com/ccaprani/bridgebeams), and it will be easy to convert to shapely. Looking forward to v2.0!
